optuna/samplers/_brute_force.py

```diff
@@ -68,23 +72,38 @@ def add_path(
             current_node = current_node.children[value]
         return current_node
 
-    def count_unexpanded(self) -> int:
+    def count_unexpanded(self, exclude_running: bool) -> int:
         # Count the number of unexpanded nodes in the subtree.
         return (
             1
-            if self.children is None
-            else sum(child.count_unexpanded() for child in self.children.values())
+            if self.children is None and (not exclude_running or not self.is_running)
+            else (
+                0
+                if self.children is None
+                else sum(
+                    child.count_unexpanded(exclude_running) for child in self.children.values()
+                )
+            )
         )
```

Comment:
A better way to implement this function is 
```
        return max(0, (
            1
            if self.children is None
            else sum(child.count_unexpanded(exclude_running) for child in self.children.values())
        ) - int(self.is_running and exclude_running))
```
This would be advantageous when there is a running trial on already-expanded node.

Reply:
Sorry, I noticed that this may cause premature stop in additional cases.